hudi-flink-datasource/hudi-flink/src/main/java/org/apache/hudi/configuration/FlinkOptions.java

```diff
@@ -712,6 +708,36 @@ private FlinkOptions() {
       .defaultValue(2)
       .withDescription("Number of partitions to list to create ClusteringPlan, default is 2");
 
+  public static final ConfigOption<Integer> CLUSTERING_PLAN_STRATEGY_SKIP_PARTITIONS_FROM_LATEST = ConfigOptions
+      .key("clustering.plan.strategy.daybased.skipfromlatest.partitions")
+      .intType()
+      .defaultValue(0)
+      .withDescription("Number of partitions to skip from latest when choosing partitions to create ClusteringPlan");
+
+  public static final ConfigOption<String> CLUSTERING_PLAN_STRATEGY_CLUSTER_BEGIN_PARTITION = ConfigOptions
+      .key("clustering.plan.strategy.cluster.begin.partition")
+      .stringType()
+      .noDefaultValue()
```

Comment:
No need to add new options for flink specifically, all the hoodie options are supported automatically, see https://github.com/apache/hudi/blob/a932e482408b72b3901404027c09e2623f5a038a/hudi-flink-datasource/hudi-flink/src/main/java/org/apache/hudi/util/FlinkWriteClients.java#L220 for details.

You can add some utilities method in `OptionsResolver` so that these option values can be fetched more easily.

Reply:
@danny0405, the `withProps(flinkConf2TypedProperties(conf))` could not support `clustering.plan.strategy.cluster.begin.partition` etc, because `clustering.plan.strategy.cluster.begin.partition` is shorter than `hoodie.clustering.plan.strategy.cluster.begin.partition`.